xbmc/filesystem/ShoutcastFile.cpp

```diff
@@ -152,14 +156,69 @@ bool CShoutcastFile::ExtractTagInfo(const char* buf)
   g_charsetConverter.wToUTF8(wConverted, strBuffer);
 
   CRegExp reTitle(true);
+  CRegExp otherData(true);
   reTitle.RegComp("StreamTitle=\'(.*?)\';");
+  otherData.RegComp("StreamUrl=\'(.*?)\';");
 
+  if (otherData.RegFind(strBuffer.c_str()) != -1)
+  {
+    std::string getOtherData(otherData.GetMatch(1));
+    if (!getOtherData.empty())
+      haveExtraData = true;
+  }
   if (reTitle.RegFind(strBuffer.c_str()) != -1)
   {
     std::string newtitle(reTitle.GetMatch(1));
     CSingleLock lock(m_tagSection);
-    result = (m_tag.GetTitle() != newtitle);
-    m_tag.SetTitle(newtitle);
+    result = (oldTitle != newtitle);
+
+    if (result && !haveExtraData)
+      m_tag.SetTitle(newtitle);
+    oldTitle = newtitle;
+
+    if (result && haveExtraData) // track has changed and extra metadata might be available
+    {
+      std::string serverUrl(otherData.GetMatch(1));
+      // if the url ends with -1 then there is no extra data to fetch
+      if ((serverUrl.find("http") != std::string::npos) &&
+          (serverUrl.find("eventdata/-1") == std::string::npos))
+      {
+        XFILE::CCurlFile http;
+        std::string extData;
+        CURL dataUrl(serverUrl);
+        http.Get(dataUrl.Get(), extData);
+
+        std::size_t titleStart = extData.find("\"eventSongTitle\":");
+        std::size_t artistStart = extData.find("\"eventSongArtist\":");
+        std::size_t imageStart = extData.find("\"eventImageUrl\":");
+        std::size_t imageEnd = extData.find("ImageUrlSmall");
+        // ensure we found some valid data
+        if (artistStart != std::string::npos && titleStart != std::string::npos)
+        {
+          titleStart += 18;
+          artistStart += 19;
+          int getArtist = imageStart - artistStart - 2;
+          int getTitle = artistStart - titleStart - 21;
+          int getCover = imageEnd - imageStart - 25;
```

Comment:
Can something be used instead of the magic numbers?

Reply:
Comment giving header structure, or url to documentation maybe.

Reply:
There is no documentation on this.  What little there is, which is all third-party, deals with just the StreamTitle part.  I have added a couple of comments with example data showing the contents of the metadata buffer and the data returned by the call to the StreamUrl.